code/obj/item/device/scanners.dm

```diff
@@ -152,19 +152,31 @@ that cannot be itched
 	var/active = 0
 	var/distancescan = 0
 	var/target = null
-	var/last_scan = "No scans have been done yet."
+
+	var/list/scans
+	var/maximum_scans = 25
+	var/number_of_scans = 0
+	var/last_scan = "No scans have been performed yet."
 
 	Topic(href, href_list)
 		..()
 		if (href_list["print"])
+			if (!(src in usr.contents))
+				boutput(usr, "<span class='notice'>You must be holding [src] that made the record in order to print it.</span>")
+				return
+			var/scan_number = text2num(href_list["print"])
+			if (scan_number < number_of_scans - maximum_scans)
+				boutput(usr, "<span class='notice'>[src] only has enough storage to hold the last [maximum_scans] scans!</span>")
```

Comment:
i don't see the point of this message

Reply:
It shows that not being able to print a record older than 25 records is intentional, rather than a bug or delay with the text window button.

Reply:
it should be an error or something then, this isn't worded like one